### PR TITLE
closes #7016 add indices to multi-pkey db schema

### LINKED ISSUE #7016: Add db index to essential columns in multi-key tables
**Indexes to be added to the schema**
- ratings
- bookshelves_books
- observations
- booknotes
- checkins

**Migration to run on ol-db1**

`ratings`
- [x] `CREATE INDEX ratings_work_id_idx ON ratings (work_id);`

`booknotes`
- [x] `CREATE INDEX booknotes_work_id_idx ON booknotes (work_id);`

`bookshelves_books`
- [x] `CREATE INDEX bookshelves_books_work_id_idx ON bookshelves_books (work_id);`

`observations`
- [x] `CREATE INDEX observations_username_idx ON observations (username);`

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #7016 please squash :)

### CODE REVIEW

File: openlibrary/core/schema.sql [hunk 1]
```diff
@@ -9,6 +8,8 @@ CREATE TABLE ratings (
     created timestamp without time zone default (current_timestamp at time zone 'utc'),
     primary key (username, work_id)
 );
+CREATE INDEX ratings_username_idx ON bookshelves_books (username);
+CREATE INDEX ratings_work_id_idx ON bookshelves_books (work_id);
```
Comment (mekarpeles) [inline openlibrary/core/schema.sql:12]:
```suggestion
CREATE INDEX ratings_work_id_idx ON ratings (work_id);
```

File: openlibrary/core/schema.sql [hunk 2]
```diff
@@ -19,6 +20,8 @@ CREATE TABLE booknotes (
     created timestamp without time zone default (current_timestamp at time zone 'utc'),
     primary key (username, work_id, edition_id)
 );
+CREATE INDEX booknotes_username_idx ON bookshelves_books (username);
+CREATE INDEX booknotes_work_id_idx ON bookshelves_books (work_id);
```
Comment (mekarpeles) [inline openlibrary/core/schema.sql:24]:
```suggestion
CREATE INDEX booknotes_work_id_idx ON booknotes (work_id);
```

File: openlibrary/core/schema.sql [hunk 3]
```diff
@@ -62,6 +66,8 @@ CREATE TABLE observations (
     created timestamp without time zone default (current_timestamp at time zone 'utc'),
     primary key (work_id, edition_id, username, observation_value, observation_type)
 );
+CREATE INDEX observations_work_id_idx ON bookshelves_books (work_id);
+CREATE INDEX observations_username_idx ON bookshelves_books (username);
```
Comment (mekarpeles) [inline openlibrary/core/schema.sql:70]:
```suggestion
CREATE INDEX observations_username_idx ON observations (username);
```

**Review by cdrini — APPROVED** [review]
Lgtm! Syntax looks right. Adding the bookshelves one in prod had quite the positive perf impact!

**Pull request merged**